include/oneapi/dpl/pstl/zip_view_impl.h

```diff
@@ -0,0 +1,505 @@
+
+        iterator_type current_;
+    }; // class iterator
+
+    template <bool Const>
+    class sentinel
+    {
+        template <typename... Sentinels>
+        constexpr explicit sentinel(const Sentinels&... sentinels) : end_(sentinels...)
+        {
+        }
+
+      public:
+        sentinel() = default;
+        constexpr sentinel(sentinel<!Const> i) requires Const &&
+            (std::convertible_to<std::ranges::sentinel_t<Views>, std::ranges::sentinel_t<__maybe_const<Const, Views>>> && ...)
+            : end_(std::move(i.end_))
+        {
+        }
+      
+        template <bool OtherConst>
+        requires(std::sentinel_for<std::ranges::sentinel_t<__maybe_const<Const, Views>>, 
+                                   std::ranges::iterator_t<__maybe_const<OtherConst, Views>>>&&...)
+        friend constexpr bool
+        operator==(const iterator<OtherConst>& x, const sentinel& y)
+        {
+            return x.compare_with_sentinels(y.end_, std::make_index_sequence<sizeof...(Views)>());
+        }
+        
+        template <bool OtherConst>
+        requires(std::sized_sentinel_for<std::ranges::sentinel_t<__maybe_const<Const, Views>>, 
+                                         std::ranges::iterator_t<__maybe_const<OtherConst, Views>>>&&...)
+        friend constexpr std::common_type_t<std::ranges::range_difference_t<__maybe_const<Const, Views>>...>        
+            operator-(const iterator<OtherConst>& x, const sentinel& y)
+        {
+            return x.distance_to_sentinels(y.end_, std::make_index_sequence<sizeof...(Views)>());
+        }
+        
+        template <bool OtherConst>
+        requires(std::sized_sentinel_for<std::ranges::sentinel_t<__maybe_const<Const, Views>>, 
+                                         std::ranges::iterator_t<__maybe_const<OtherConst, Views>>>&&...)
+        friend constexpr std::common_type_t<std::ranges::range_difference_t<__maybe_const<Const, Views>>...>
+            operator-(const sentinel& y, const iterator<OtherConst>& x)
+        {
+            return -(x - y);
+        }
+
+      private:
+        friend class zip_view;
+        
+        tuple_type<std::ranges::sentinel_t<__maybe_const<Const, Views>>...> end_;
+    }; // class sentinel
+
+    constexpr auto
+    begin() requires(!(__simple_view<Views> && ...))
+    {
+        using iterator_type = tuple_type<std::ranges::iterator_t<__maybe_const<false, Views>>...>;
+        auto __tr = [](auto&&... __args) {
+            return iterator<false>(iterator_type(std::forward<decltype(__args)>(__args)...)); 
+        };
+        return apply_to_tuple(__tr, std::ranges::begin, views_);
+    }
+
+    constexpr auto
+    begin() const requires(std::ranges::range<const Views>&&...)
+    {
+        using iterator_type = tuple_type<std::ranges::iterator_t<__maybe_const<true, Views>>...>;
+        auto __tr = [](auto&&... __args) { 
+            return iterator<true>(iterator_type(std::forward<decltype(__args)>(__args)...)); 
```

Comment:
Why do we need to convert to tuple explicitly and cannot construct `iterator<true>` directly from the set of begin iterators?

Reply:
I did such way because `std::zip_view` C++23 uses the same approach.
Before, when we had "construct iterator<true> directly from the set of begin iterators", several tests were failed.
(Unfortunately, I don't remember which issues were.)

Reply:
Oh, I see now. In that case, I believe an alias `iterator_type` is not the best name for tuple of iterators. 
Because it misleads the code reader (here and in the constructor of iterator/sentinel).

Reply:
So, please propose the best name :-)
I can propose only `current_type` that was before)
`tuple_tuple|` is already used for OneDPL internal tuple.

Reply:
`iterator_tuple_type` is too long...
M.b. `tuple_data`? or just `iterators`? BTW, `iterators` name at least reflects sense of this member :-)